rele/client.py

```diff
@@ -31,24 +31,19 @@ class Subscriber:
     For convenience, this class wraps the creation and consumption of a topic
     subscription.
 
-    :param gc_project_id: string Google Cloud Project ID.
-    :param credentials: string Google Cloud Credentials.
+    :param gc_project_id: string :ref:`_settings_project_id` .
+    :param credentials: obj :meth:`~rele.config.Config.credentials`.
     :param default_ack_deadline: int Ack Deadline defined in settings
     """
 
-    def __init__(self, gc_project_id=None, credentials=None, default_ack_deadline=None):
-
-        if gc_project_id is None or credentials is None:
-            creds, project = get_google_defaults()
-
-        self._gc_project_id = gc_project_id or project
+    def __init__(self, gc_project_id, credentials, default_ack_deadline=None):
```

Comment:
if we are deprecating it, this parameter should stay optional

Reply:
We are deprecating it from the settings. However from the project, like the subscriber client, I am afraid we can not remove it. 
I did decide to remove the non-default values here since we are no longer taking the credentials from the environment in the client. Instead we take the values from the environment in the config.

Reply:
Basically, instead of having credentials being instantiated in multiple places, I have consolidated the handling of credentials in the Config class, no where else.